indexer/services/ender/src/handlers/stateful-order/stateful-order-replacement-handler.ts

```diff
@@ -0,0 +1,73 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  OrderTable,
+} from '@dydxprotocol-indexer/postgres';
+import { getOrderIdHash } from '@dydxprotocol-indexer/v4-proto-parser';
+import {
+  IndexerOrder,
+  IndexerOrderId,
+  OffChainUpdateV1,
+  OrderPlaceV1_OrderPlacementStatus,
+  StatefulOrderEventV1,
+} from '@dydxprotocol-indexer/v4-protos';
+import * as pg from 'pg';
+
+import config from '../../config';
+import { ConsolidatedKafkaEvent } from '../../lib/types';
+import { AbstractStatefulOrderHandler } from '../abstract-stateful-order-handler';
+
+export class StatefulOrderReplacementHandler
+  extends AbstractStatefulOrderHandler<StatefulOrderEventV1> {
+  eventType: string = 'StatefulOrderEvent';
+
+  private getOrderId(): string {
+    const orderId = OrderTable.orderIdToUuid(this.event.orderReplacement!.order!.orderId!);
+    return orderId;
+  }
+
+  public getParallelizationIds(): string[] {
+    // Stateful Order Events with the same orderId
+    return this.getParallelizationIdsFromOrderId(this.getOrderId());
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  public async internalHandle(resultRow: pg.QueryResultRow): Promise<ConsolidatedKafkaEvent[]> {
+    const oldOrderId = this.event.orderReplacement!.oldOrderId!;
+    const order = this.event.orderReplacement!.order!;
+    if (resultRow.errors != null) {
+      logger.error({
```

Comment:
nit: the next time an error is added to that postgres script, this can break. Add a check indicating this error is actually an order replacement related error

Reply:
Added a comment and the test should catch if the error at index 0 is no longer the expected error

Reply:
Updated to check if error message is related to order replacement!